seed/src/lib.rs

```diff
@@ -270,30 +270,50 @@ impl Node {
     ) -> Result<(), Error> {
         let peer_id = peer_info.peer_id;
         let url = urn.clone().into_rad_url(peer_id);
+        let proejct_urn = RadUrn {
+            path: uri::Path::new(),
+            ..urn.clone()
+        };
         let port = peer_info.advertised_info.listen_port;
         let addr_hints = peer_info
             .seen_addrs
             .iter()
             .map(|a: &std::net::IpAddr| (*a, port).into())
             .collect::<Vec<_>>();
 
-        let result = {
+        // FIXME(xla): There should be a saner way to test.
```

Comment:
I hear you, and am in the process of removing the distinction between `clone` and `fetch`.

That being said, you should be able to condense the code to track (unconditionally), then either fetch or clone depending on the result of `has_urn`.